main.go

```diff
@@ -109,54 +98,90 @@ func getInterval() (int, error) {
 	return integerGithubAPIInterval, nil
 }
 
-func getECRImageScanFindings(repositories []string) ([]findingsInfo, error) {
+func getECRImageScanFindings(repositories []string) error {
 	sess := session.Must(session.NewSessionWithOptions(session.Options{
 		SharedConfigState: session.SharedConfigEnable,
 	}))
 
 	svc := ecr.New(sess)
-	findingsInfos := []findingsInfo{}
-	results := []findingsInfo{}
 
 	imageTags, err := getImageTags()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get image tags: %w", err)
+		return fmt.Errorf("failed to get image tags: %w", err)
 	}
 
+	var wg sync.WaitGroup
+
 	for _, repo := range repositories {
 		for _, imageTag := range imageTags {
-			input := &ecr.DescribeImageScanFindingsInput{
-				ImageId:        &ecr.ImageIdentifier{ImageTag: aws.String(imageTag)},
-				RepositoryName: aws.String(repo),
-			}
+			wg.Add(1)
 
-			for {
-				findings, err := svc.DescribeImageScanFindings(input)
-				//nolint:gocritic,errorlint
-				if aerr, ok := err.(awserr.Error); ok {
-					switch aerr.Code() {
-					case "ScanNotFoundException":
-						log.Printf("Skip the repository %v with imageTag %v. %v\n", repo, imageTag, err.Error())
-					case "ImageNotFoundException":
-						log.Printf("Skip the repository %v with imageTag %v. %v\n", repo, imageTag, err.Error())
-					default:
-						return nil, fmt.Errorf("failed to describe image scan findings: %w", err)
-					}
-				} else if findings.ImageScanFindings == nil {
-					log.Printf("Skip the repository %v with imageTag %v. ImageScanStatus: Status %v Description %v\n", repo, imageTag, findings.ImageScanStatus.Status, findings.ImageScanStatus.Description)
-				} else {
-					results = generateFindingsInfos(findings, imageTag, repo)
+			go func(svc *ecr.ECR, repo string, imageTag string) {
+				result, err := describeImageScanFindings(svc, repo, imageTag)
+				if err != nil {
+					log.Printf("failed to describe image scan findings: %v", err)
```

Comment:
The error should be wrapped and return to a caller side but I'm not sure how to do it.
I'll revisit later.